### PR TITLE
Maya: aov filtering

### PR DESCRIPTION
**Pull request by moonyuet**
## Brief description
fixing the bug for the regex filter not excluding the aov in RedShift
See(https://github.com/pypeclub/OpenPype/issues/4057)

## Description
Multipart boolean option of the renderers (RedShift and Arnold) is broken in lib_renderproduct.py, which submits a wrong information to collect_render.py(i.e. multipart is always True even when it is being turned off..), This cause the review option not working with regex filters, which depends on whether multipart is on /off.

For this fix, 
if “force combine beauty and AOVs into single file” or “multipart EXR” in Redshift Render Setting is off, it will filter out the renderpass according to the regex filter set in the Openpype Setting. (I set .*([Bb]eauty).* as filter)
image

If either one is on, it will show both beauty and Cryptomatte pass.

image

If mergeAOVs or multipart in Arnold Render Setting is off, it will filter out the renderpass in regards to the regex filter set in the Openpype setting. .

image

 If multipart is on, it will render all the passes for the multipart option (I am not sure if it’s a good idea. so let me know if you also want the filter option included here.) 
image

while it will just only render beauty pass for mergeAOVs
image

## Additional info
No Additional Info

## Testing Notes
1. check if you have disable the "multipart" or "force combine beauty and AOVs" in Redshift Render Setting  OR
   check if you have disable the "mutlipart" or "Merge AOVs" in Arnold Render Setting.
image
image

2. set your regex filter in deadline setting
image
3. add the AOV maps
image
4. Create the render instance and run the publisher

### CODE REVIEW

**Comment by m-u-r-p-h-y** [comment]
the render settings under `project_settings/maya/RenderSettings` should only be used as render presets, not for evaluation during rendering. We should always take the settings from the render scene itself.

**Comment by moonyuet** [comment]
> the render settings under `project_settings/maya/RenderSettings` should only be used as render presets, not for evaluation during rendering. We should always take the settings from the render scene itself.

I find out that the boolean of multipart in ` collect_render.py` doesn't align with that of multipart in `lib_renderproduct.py`, I have edited the `lib_renderproduct.py` so that the multipart works properly in Redshift. For Arnold, I still want to see if you guys want to have the filtering even if the multipart is on  as it seems that it still renders all passes when the multipart is on.

**Review by m-u-r-p-h-y — DISMISSED** [review]
tested the redshift and there is a problem with the code which was taking care of Redshift change of `Beauty` name to` BeautyAux`


image

you can see actually rendered files and the content of the JSON file with the wrong expected files

**Review by LiborBatek — APPROVED** [review]
So now it works also in Redshift. Seems ok to me. Not spotted any problems.

File: openpype/hosts/maya/api/lib_renderproducts.py [hunk 1]
```diff
@@ -1044,7 +1054,7 @@ def get_render_products(self):
 
             # Any AOVs that still get processed, like Cryptomatte
             # by themselves are not multipart files.
-            aov_multipart = not multipart
+            # aov_multipart = not multipart
```
Comment (antirotor) [inline openpype/hosts/maya/api/lib_renderproducts.py:1057]: I would just remove this

File: openpype/modules/deadline/plugins/publish/submit_publish_job.py [hunk 2]
```diff
@@ -494,12 +494,13 @@ def _create_instances_for_aov(self, instance_data, exp_files):
...
             aov_patterns = self.aov_filter
-            preview = match_aov_pattern(app, aov_patterns, render_file_name)
 
+            preview = match_aov_pattern(app, aov_patterns, render_file_name)
             # toggle preview on if multipart is on
+
             if instance_data.get("multipartExr"):
                 preview = True
-
+            self.log.info("preview:{}".format(preview))
```
Comment (antirotor) [inline openpype/modules/deadline/plugins/publish/submit_publish_job.py:503]:
```suggestion
            self.log.debug("preview:{}".format(preview))
```

File: openpype/modules/deadline/plugins/publish/submit_publish_job.py [hunk 3]
```diff
@@ -542,7 +543,7 @@ def _create_instances_for_aov(self, instance_data, exp_files):
             if new_instance.get("extendFrames", False):
                 self._copy_extend_frames(new_instance, rep)
             instances.append(new_instance)
-
+            self.log.info("instances:{}".format(instances))
```
Comment (antirotor) [inline openpype/modules/deadline/plugins/publish/submit_publish_job.py:546]:
```suggestion
            self.log.debug("instances:{}".format(instances))
```

**Review by antirotor — COMMENTED** [review]
Just some minor cosmetic changes

**Review by antirotor — APPROVED** [review]

**Pull request merged**